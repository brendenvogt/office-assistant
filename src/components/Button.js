import React from 'react';
import {
  Button,
} from 'react-native-ui-lib';
import { colors, fonts } from '../styles';

export default function CustomButton(props) {
  return (
    <Button
      {...props}
      backgroundColor={colors.red}
      labelStyle={{
        fontFamily: fonts.primary,
        fontWeight: 'bold',
      }}
      style={{
        paddingVertical: 10,
        paddingHorizontal: 50,
        ...props.style,
      }}
    />
  );
}
