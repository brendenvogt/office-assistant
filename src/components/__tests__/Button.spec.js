/* eslint-disable no-undef */
import React from 'react';
import Enzyme from 'enzyme';

import Button from '../Button';

describe('Testing ToggleButton component', () => {
  it('renders as expected', () => {
    const wrapper = Enzyme.shallow(
      <Button
        label="Hello"
        onPress={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
