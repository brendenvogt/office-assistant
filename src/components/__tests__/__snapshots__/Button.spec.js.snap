// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ToggleButton component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomButton
    label="Hello"
    onPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColor": "#e32d2f",
      "iconOnRight": false,
      "label": "Hello",
      "labelStyle": Object {
        "fontFamily": "Muli",
        "fontWeight": "bold",
      },
      "onPress": [Function],
      "outline": false,
      "size": "large",
      "style": Object {
        "paddingHorizontal": 50,
        "paddingVertical": 10,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColor": "#e32d2f",
        "iconOnRight": false,
        "label": "Hello",
        "labelStyle": Object {
          "fontFamily": "Muli",
          "fontWeight": "bold",
        },
        "onPress": [Function],
        "outline": false,
        "size": "large",
        "style": Object {
          "paddingHorizontal": 50,
          "paddingVertical": 10,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
